src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -41,24 +41,30 @@ internal sealed partial class DotNetCoreProjectCompatibilityDetector : IDotNetCo
         private readonly Lazy<IVsShellUtilitiesHelper> _shellUtilitiesHelper;
         private readonly Lazy<IFileSystem> _fileSystem;
         private readonly Lazy<IHttpClient> _httpClient;
+        private readonly Lazy<IProjectCreationInfoService> _projectCreationInfoService;
         private readonly IVsService<ISettingsManager> _settingsManagerService;
         private readonly IVsService<IVsUIShell> _vsUIShellService;
         private readonly IVsService<IVsSolution> _vsSolutionService;
         private readonly IVsService<IVsAppId> _vsAppIdService;
         private readonly IVsService<IVsShell> _vsShellService;
         private RemoteCacheFile _versionDataCacheFile;
-        private Version _ourVSVersion;
         private uint _solutionCookie = VSConstants.VSCOOKIE_NIL;
-        private bool _solutionOpened;
-        private CompatibilityLevel _compatibilityLevelWarnedForThisSolution = CompatibilityLevel.Recommended;
         private DateTime _timeCurVersionDataLastUpdatedUtc = DateTime.MinValue; // Tracks how often we meed to look for new data
-        private VersionCompatibilityData _curVersionCompatibilityData;
         private IVsSolution _vsSolution;
 
+        // These are internal for unit testing
+        internal bool SolutionOpen { get; private set; }
```

Comment:
these properties where added so tests could introspect into what is happening in this class.  I hope to remove them someday after this class has been refactored